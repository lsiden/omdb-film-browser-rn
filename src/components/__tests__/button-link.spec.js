import React from "react"
import renderer from "react-native-test-utils"

import ButtonLink from "components/button-link"

let onPress

const defaultProps = () => ({
  onPress,
})

const buttonContents = "Push Me"

const createWrapper = (props = {}) => {
  onPress = jest.fn()
  return renderer(
    <ButtonLink {...{ ...defaultProps(), ...props }}>
      {buttonContents}
    </ButtonLink>
  )
}

it("renders children", () => {
  const wrapper = createWrapper()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

test("click event invokes onPress()", () => {
  const wrapper = createWrapper()
  wrapper.simulate("press")
  expect(onPress).toHaveBeenCalled()
})
