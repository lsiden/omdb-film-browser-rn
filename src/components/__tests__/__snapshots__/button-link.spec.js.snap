// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children 1`] = `
<button
  onPress={[MockFunction]}
  style={
    Object {
      "background": "none",
      "border": "none",
      "cursor": "pointer",
      "fontSize": "12pt",
      "padding": "0!important",
    }
  }
>
  Push Me
</button>
`;
